src/client/workflow/context.ts

```diff
@@ -178,39 +127,41 @@ export class WorkflowContext<TInitialPayload = unknown> {
    * @param initHeaderValue Whether the invocation should create a new workflow
    * @returns
    */
-  public getHeaders(initHeaderValue: "true" | "false") {
-    return {
+  public getHeaders(initHeaderValue: "true" | "false", step?: Step, contentType?: string) {
+    let baseHeaders: Record<string, string> = {
       [WORKFLOW_INIT_HEADER]: initHeaderValue,
       [WORKFLOW_ID_HEADER]: this.workflowId,
       [`Upstash-Forward-${WORKFLOW_PROTOCOL_VERSION_HEADER}`]: WORKFLOW_PROTOCOL_VERSION,
     };
-  }
-
-  /**
-   * Creates a workflow from a request by parsing the body and checking the
-   * headers.
-   *
-   * @param request request received from the API
-   * @param client QStash client
-   * @param initialPayloadParser function to parse the initial payload
-   * @returns workflow and whether its the first time the workflow is being called
-   */
-  static async createContext<TInitialPayload = unknown>(
-    request: Request,
-    client: Client,
-    initialPayloadParser: InitialPayloadParser<TInitialPayload>
-  ) {
-    const { isFirstInvocation, workflowId, initialPayload, steps } =
-      await WorkflowContext.parseRequest(request);
 
-    const workflowContext = new WorkflowContext<TInitialPayload>({
-      client,
-      workflowId,
-      initialPayload: initialPayloadParser(initialPayload),
-      steps,
-      url: request.url,
-    });
+    if (step?.callHeaders && step.callMethod) {
+      const forwardedHeaders = Object.fromEntries(
+        Object.entries(step.callHeaders).map(([header, value]) => [
+          `Upstash-Forward-${header}`,
+          value,
+        ])
+      );
+      baseHeaders = {
```

Comment:
Why do we reassign? Can't we just mutate the object?

Reply:
I removed the reassign. I moved the getHeaders from public API to workflow-requests.ts too. [1ef8619](https://github.com/upstash/qstash-js/pull/132/commits/1ef8619a61926ac6108b2c44653638ae219908c0)